distribution/ddf/src/main/filtered-resources/etc/org.apache.karaf.features.cfg

```diff
@@ -22,41 +22,60 @@
  mvn:ddf.security/security-services-app/${project.version}/xml/features, \
+ mvn:ddf.admin/admin-app/${project.version}/xml/features, \
+ mvn:ddf.distribution/install-profiles/${project.version}/xml/features, \
  mvn:org.codice.ddf.broker/broker-app/${project.version}/xml/features, \
  mvn:ddf.catalog/catalog-app/${project.version}/xml/features, \
  mvn:ddf.catalog.solr/solr-app/${project.version}/xml/features, \
  mvn:org.codice.opendj.embedded/opendj-embedded-app/${opendj-embedded.app.version}/xml/features, \
  mvn:org.codice.ddf.spatial/spatial-app/${project.version}/xml/features, \
  mvn:ddf.ui/search-ui-app/${project.version}/xml/features, \
- mvn:ddf.admin/admin-app/${project.version}/xml/features, \
  mvn:org.codice.ddf.resourcemanagement/resourcemanagement-app/${project.version}/xml/features, \
  mvn:org.codice.ddf.spatial/geowebcache-app/${project.version}/xml/features, \
- mvn:org.codice.ddf.registry/registry-app/${project.version}/xml/features, \
- mvn:org.apache.karaf.features/enterprise/${karaf.version}/xml/features
+ mvn:org.codice.ddf.registry/registry-app/${project.version}/xml/features
 
 #
 # Comma separated list of features to install at startup
+#
+# The parentheses in the featuresBoot list below is used to install sets of features in stages. See
+# https://github.com/apache/karaf/blob/karaf-4.1.1/features/core/src/main/java/org/apache/karaf/features/internal/service/BootFeaturesInstaller.java#L105.
+# The stages enforce an order of installation of features.
+#
+# This staged installation of these featuresBoot features provides another benefit besides
+# eliminating the need for some prerequisite="true" flags. Without specifying these stages, all
+# featuresBoot features were considered to be in the same stage, and they would all be installed at
+# once. This means that if one feature failed to be installed, all of the featuresBoot features
+# would fail to be installed. In this new configuration that includes the multi-stage installation,
+# each stage is installed one after the other. This proves to be useful, especially, for the
+# (kernel,ddf-branding) step which starts the karaf console. For example, if platform-app failed to
+# be installed, kernel and ddf-branding would already be installed. The Karaf console, which is
+# started in the kernel feature, could be used to investigate why the platform-app failed.
+#
+# This multi-stage installation of the featuresBoot features is not documented well, but it appears
+# to work for the current feature files included in DDF and in the itests. This configuration change
+# to the DDF featuresBoot was made in the Karaf upgrade to 4.1.1.
+#
+# TODO DDF-3072 Investigate improvements of this configuration.
 featuresBoot= \
- kernel, \
- ddf-branding, \
- platform-app, \
- security-services-app, \
- admin-app, \
- admin-modules-installer
+ (kernel), \
+ (ddf-branding), \
+ (platform-app), \
+ (security-services-app), \
+ (admin-app, \
+ admin-modules-installer),,
```

Comment:
I made a comment regarding this on the ticket to "Investigate configuration of featuresBoot using a staged installation":
https://codice.atlassian.net/browse/DDF-3072?focusedCommentId=30302&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-30302.